Source/dom/layers/StyledLayer.js

```diff
@@ -15,19 +17,73 @@ Factory.registerClass(StyledLayer, MSImmutableStyledLayer)
 
 StyledLayer.define('style', {
   get() {
-    const style = Style.fromNative(this._object.style())
-    return style
+    return Style.fromNative(this._object.style())
   },
   set(style) {
     if (this.isImmutable()) {
       return
     }
+
+    // we can then actually set the style
+    let nativeStyle
     if (isNativeObject(style)) {
       this._object.style = style
     } else if (!style || !style.sketchObject) {
       this._object.style = new Style(style).sketchObject
     } else {
       this._object.style = style.sketchObject
     }
+
+    this._object.style = nativeStyle.copy()
+  },
+})
+
+StyledLayer.define('sharedStyleId', {
+  get() {
+    const nativeSharedStyle = this._object.sharedStyleID()
+    if (!nativeSharedStyle) {
+      return null
+    }
+    return String(nativeSharedStyle)
+  },
+  set(sharedStyleId) {
+    if (this.isImmutable()) {
+      return
+    }
+
+    if (!sharedStyleId) {
+      this._object.setSharedStyleID(null)
+      return
+    }
+
+    this._object.setSharedStyleID(sharedStyleId)
+  },
+})
+
+StyledLayer.define('sharedStyle', {
+  enumerable: false,
+  exportable: false,
+  get() {
+    if (!this._object.sharedObject) {
+      return null
+    }
+    const nativeSharedStyle = this._object.sharedObject()
+    if (!nativeSharedStyle) {
+      return null
+    }
+    return SharedStyle.fromNative(nativeSharedStyle)
+  },
+  set(sharedStyle) {
+    if (this.isImmutable()) {
+      return
+    }
+
+    if (!sharedStyle) {
+      this._object.setSharedStyleID(null)
+      return
+    }
+
+    const nativeSharedStyle = wrapObject(sharedStyle)
+    this._object.setSharedStyleID(nativeSharedStyle.id)
```

Comment:
You said setting the sharedStyle won't update the style but will it set the sharedStyleID on the layer and the sharedStyleID of the style? Hard to tell without what is happening internally when this is called. In other words, is the following always true after setting any of the (shared)style properties? `layer.style.sharedId == layer.sharedStyleID == layer.sharedStyle`

Reply:
A style doesn’t have sharedStyleId anymore. It is not linked to SharedStyle in any way

Reply:
Any reason to not apply the style here? I can't think of a reason you wouldn't want to do both. If not, any thoughts on adding a helper like Style used to have? So you could replace:
```
layer.sharedStyle = aSharedStyle
layer.style.syncWithSharedStyle(layer.sharedStyle)
```
with
```
layer.sharedStyle = aSharedStyle
layer.syncWithSharedStyle()
```